demo/grid-columns-demos.html

```diff
@@ -6,6 +6,62 @@
       }
     </style>
 
+    <h3>Automatic Column Width</h3>
+    <p>
+      Applying <code>auto-width</code> on the <code>&lt;vaadin-grid-column&gt;</code> will set the width of the column
+      to automatically fit the content.
+    </p>
+    <p>
+      The column width will be calculated and set only once based on the contents of the initially rendered cells.
+      You can manually trigger the auto sizing behavior again by calling <code>grid.recalculateColumnWidths()</code>.
+    </p>
+    <p>
+      By default the column might still expand if there is more space available, unless you also set <code>flex-grow="0"</code>.
+    </p>
+    <p>
+      <b>Note:</b> If all the columns have <code>flex-grow="0"</code> there might be empty space after the last column
+      if the grid is wider than what is needed to fit all the columns. You probably want to leave at least one column
+      without <code>flex-grow="0"</code> so that column will take all the remaining space that is left.
+    </p>
+    <vaadin-demo-snippet id='grid-columns-demos-auto-width'>
+      <template preserve-content>
+        <vaadin-button>Recalculate column widths</vaadin-button>
+        <vaadin-grid theme="compact column-borders" aria-label="Automatic Column Width Example">
+          <vaadin-grid-column auto-width flex-grow="0" text-align="end" header="#"></vaadin-grid-column>
```

Comment:
Using auto-width on this column doesn't look good in the demo as indexes after 99 become clipped and show the ellipsis.

Reply:
I was thinking about setting a fixed width for this to fit those numbers, but I thought that's a good case for testing the "Recalculate column widths" button. But I agree it looks quite ugly. I'll change this column to fixed width. There is still some content which will overflow when you scroll so you can see the effect of the button too.